src/main/java/org/openrewrite/maven/ResourceParser.java

```diff
@@ -3,107 +3,168 @@
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+                if (!attrs.isOther() && !alreadyParsed.contains(file) && !isIgnored(file)) {
+                    if (isOverSizeThreshold(attrs.size())) {
+                        logger.info("Parsing as Quark " + file + " as its size + " + attrs.size() / (1024L * 1024L) +
+                                "Mb exceeds size threshold " + sizeThresholdMb + "Mb");
+                        quarkPaths.add(file);
+                    } else {
+                        resources.add(file);
+                    }
+                }
+                return FileVisitResult.CONTINUE;
             }
+        });
 
-            if (alreadyParsed.contains(path)) {
-                return false;
-            }
+        List<S> sourceFiles = new ArrayList<>(resources.size() + quarkPaths.size());
 
-            for (String exclusion : exclusions) {
-                PathMatcher matcher = baseDir.getFileSystem().getPathMatcher("glob:" + exclusion);
-                if (matcher.matches(baseDir.relativize(path))) {
-                    alreadyParsed.add(path);
-                    return false;
-                }
-            }
+        JsonParser jsonParser = new JsonParser();
+        List<Path> jsonPaths = new ArrayList<>();
+
+        XmlParser xmlParser = new XmlParser();
+        List<Path> xmlPaths = new ArrayList<>();
+
+        YamlParser yamlParser = new YamlParser();
+        List<Path> yamlPaths = new ArrayList<>();
+
+        PropertiesParser propertiesParser = new PropertiesParser();
+        List<Path> propertiesPaths = new ArrayList<>();
+
+        ProtoParser protoParser = new ProtoParser();
+        List<Path> protoPaths = new ArrayList<>();
+
+        HclParser hclParser = HclParser.builder().build();
+        List<Path> hclPaths = new ArrayList<>();
 
-            long fileSize = attrs.size();
-            if ((sizeThresholdMb > 0 && fileSize > sizeThresholdMb * 1024L * 1024L)) {
-                alreadyParsed.add(path);
-                logger.info("Skipping parsing " + path + " as its size + " + fileSize / (1024L * 1024L) +
-                        "Mb exceeds size threshold " + sizeThresholdMb + "Mb");
-                return false;
+        QuarkParser quarkParser = new QuarkParser();
+
+        resources.forEach(path -> {
+            if (jsonParser.accept(path)) {
+                jsonPaths.add(path);
+            } else if (xmlParser.accept(path)) {
+                xmlPaths.add(path);
+            } else if (yamlParser.accept(path)) {
+                yamlPaths.add(path);
+            } else if (propertiesParser.accept(path)) {
+                propertiesPaths.add(path);
+            } else if (protoParser.accept(path)) {
+                protoPaths.add(path);
+            } else if (hclParser.accept(path)) {
+                hclPaths.add(path);
+            } else if (quarkParser.accept(path)) {
+                quarkPaths.add(path);
```

Comment:
Java files that do not exist in a maven module or are in a module but exist outside of `main/test` are added as quarks. We could use the JavaParser to parse those as resources if preferred.